docs/DeveloperGuide.adoc

```diff
@@ -297,58 +298,162 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |clerk|find out the upcoming appointments for given patients. |
+
+|`* * *` |clerk|update the doctors' details by typing commands and user details. |
+
+|`* * *` |clerk|add new doctors into system |
+
+|`* *` |clerk|remove a patient from the queue if they leave. |
+
+|`* *` |clerk|view the number of patients who visited the clinic today |
+
+|`* * *` |clerk|edit patients' details | keep their particulars up to date
+
+|`* * *` |clerk|register new patients with optional fields |
+
+|`* *` |clerk|find all patients who have missed their appointments. | keep track of the list of patients whom I need to inform
+
+|`* * *` |clerk|add ad-hoc patients to the queue |
+
+|`* *` |clerk|acknowledge appointments if patients are present for their appointments | keep track of patients who came for their appointments
+
+|`* *` |clerk|tag patient with known allergies. | keep track of their allegies
+
+|`* *` |clerk|cancel appointments for patients | free up appointment time slots
+
+|`* * *` |clerk|search for patients using their name or phone number |
+
+|`* * *` |clerk|look up how many patients are in the queue, on a side panel | recommend estimated time that the patient will be attended to
+
+|`* * *` |clerk|look up patient using a reference id. |
+
+|`* * *` |clerk|reschedule appointments of patients |
+
+|`* * *` |clerk|search for appointment slots easily | schedule appointments for patients easily.
+
+|`* * *` |clerk|assign a queue number to each patient in the queue |
+
+|`* *` |clerk|schedule patient's follow up appointments |
+
+|`* * *` |clerk|use the appointment scheduler | schedule appointments for my patients
+
+|`* * *` |clerk|save time managing the queue | have more time to do my own work
+
+|`* * *` |clerk|take note of the doctors that are on-shift | effectively direct patients to available doctors.
+
+|`* *` |experienced user|create custom aliases to commands |
+
+|`* *` |receptionist|use the undo and redo feature to remedy any mistakes.|
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
 
 _{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `ClerkPro` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Add patient into queue (UC1)
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  New patient arrives at the clinic
+2.  User wants to add new patient into the queue
+3.  ClerkPro adds the patient into the queue
+
+Use case ends.
+
+*Extensions*
+
+* 2a. Actor inputs invalid format
```

Comment:
should use consistent terms to refer to the same role. User and Actor both seem to refer to the same person